src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -638,13 +643,14 @@ Subgraph::DataFlowPasses Subgraph::getDataFlowPasses() {
     SNIPPETS_REGISTER_PASS_ABSOLUTE_COMMON(Place::PipelineStart, ConvertToSwishCPU);
     SNIPPETS_REGISTER_PASS_RELATIVE_COMMON(Place::After, ov::snippets::pass::Canonicalization,
                                            ov::snippets::pass::AnalyzeBroadcastableInputs, broadcastable_inputs);
-    if (context->getConfig().inferencePrecision == ov::element::bf16 && subgraph_attrs->snippet->has_domain_sensitive_ops()) {
+    if ((context->getConfig().inferencePrecision == ov::element::bf16 || context->getConfig().inferencePrecision == ov::element::f16)
+         && subgraph_attrs->snippet->has_domain_sensitive_ops()) {
         // enforce BF16 precisions to supported operations
         // MatMul has to be decomposed to Brgemm operations before enforcement
         // Note, MatMul decomposition will be run later again for case if BF16 enforcement is not happened
         SNIPPETS_REGISTER_PASS_ABSOLUTE_X86_64(Place::PipelineStart, ov::snippets::pass::MatMulToBrgemm);
         SNIPPETS_REGISTER_PASS_RELATIVE_X86_64(Place::After, ov::snippets::pass::MatMulToBrgemm,
-                                               pass::EnforcePrecision, element::f32, element::bf16);
+                                               pass::EnforcePrecision, element::f32, context->getConfig().inferencePrecision);
```

Comment:
Looks like we should update the default behavior of this pass too. Could you please cover `f16` case in the pass `EnforcePrecision::get_supported_precisions_default`?
https://github.com/openvinotoolkit/openvino/blob/a80aea6c3e27e83c5b0bd6c35f253514dd1c4eb1/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/enforce_precision.cpp#L123-L128

Reply:
done, thanks Alexandra!